.github/workflows/acceptance-tests.yml

```diff
@@ -39,58 +39,42 @@ jobs:
           distribution: 'temurin'
           cache: 'gradle'
       - name: Validate Gradle wrapper
-        uses: gradle/wrapper-validation-action@v1
+        uses: gradle/actions/wrapper-validation@v3
       - name: Acceptance tests in sandbox
         env:
           TL_CLIENT_ID: ${{ secrets.tl_client_id }}
           TL_CLIENT_SECRET: ${{ secrets.tl_client_secret }}
           TL_SIGNING_KEY_ID: ${{ secrets.tl_signing_key_id }}
           TL_SIGNING_PRIVATE_KEY: ${{ secrets.tl_signing_private_key }}
         run: ./gradlew acceptance-tests
+      - uses: actions/upload-artifact@v4
+        name: Upload test results as artifact
+        if: always()
+        with:
+          name: test-results
+          path: ./build/test-results
+          retention-days: 1
   notify-slack:
     needs: acceptance-tests
     runs-on: ubuntu-latest
     if: ${{ always() && contains(needs.*.result, 'failure') && inputs.notify_slack_on_failure == true }}
     steps:
-      - name: Notify slack
-        uses: slackapi/slack-github-action@v1.24.0
+      - uses: actions/download-artifact@v4
+        name: Download test results artifacts
+        with:
+          name: test-results
+          path: ./build/test-results
+      - uses: dili91/testvox@v0.1.0
+        name: Generate Slack report from Junit results
```

Comment:
This is public library/action I've worked on recently. [Link](https://github.com/marketplace/actions/testvox-turn-test-results-into-human-readable-reports)